tests/routes/report.test.js

```diff
@@ -0,0 +1,135 @@
+      chai.request(server)
+        .post(`/api/v1/report/articles/${Slug}`)
+        .set('Authorization', `Bearer: ${accessToken}`)
+        .send({
+          reportType: 'plagiarism',
+          description: 'gsgsgsgsgsgggssgsgsggsgsgsgsg'
+        })
+        .end((err, res) => {
+          res.should.have.status(200);
+          res.body.should.have.property('status');
+          res.body.should.have.property('message');
+          done();
+        });
+    });
+    it('should not be able to report an article without type or description', (done) => {
+      chai.request(server)
+        .post(`/api/v1/report/articles/${Slug}`)
+        .set('Authorization', `Bearer: ${accessToken}`)
+        .send({})
+        .end((err, res) => {
+          res.should.have.status(400);
+          res.body.should.have.property('status');
+          res.body.should.have.property('message');
+          done();
+        });
+    });
+    it('should  not be able to report articles not existing', (done) => {
+      chai.request(server)
+        .post(`/api/v1/report/articles/${Slug}`)
+        .set('Authorization', 'Bearer: wrongtoken')
+        .send({})
+        .end((err, res) => {
+          res.should.have.status(401);
+          res.body.should.have.property('status');
+          res.body.should.have.property('message');
+          done();
+        });
+    });
+    it('should  not be able to report articles not existing', (done) => {
+      chai.request(server)
+        .post('/api/v1/report/articles/not-existing-slug')
+        .set('Authorization', `Bearer: ${accessToken}`)
+        .send({
+          reportType: 'plagiarism',
+          description: 'gsgsgsgsgsgggssgsgsggsgsgsgsg'
+        })
+        .end((err, res) => {
+          res.should.have.status(404);
+          res.body.should.have.property('status');
+          res.body.should.have.property('message');
+          done();
+        });
+    });
+    it('should  not be able to report articleswhen reportType value is wrong', (done) => {
+      chai.request(server)
+        .post(`/api/v1/report/articles/${Slug}`)
+        .set('Authorization', `Bearer: ${accessToken}`)
+        .send({
+          reportType: 'wrongValue',
+          description: 'gsgsgsgsgsgggssgsgsggsgsgsgsg'
+        })
+        .end((err, res) => {
+          res.should.have.status(400);
+          res.body.should.have.property('status');
+          res.body.should.have.property('message');
+          done();
+        });
+    });
+    it('should returns articles bookmarked by the author', (done) => {
+      chai.request(server)
+        .get('/api/v1/report/articles?page=1&size=3')
+        .set('Authorization', `Bearer: ${accessToken}`)
+        .end((err, res) => {
+          res.should.have.status(200);
+          res.body.should.have.property('status');
+          res.body.should.have.property('message');
+          res.should.be.an('object');
+          res.body.data.reports.article.should.be.an('array');
```

Comment:
Kindly check this again, it seems there is no 'article' property in the 'reports' property. This test is failing because of this. The 'reports' seems to be containing the array.

Reply:
Okay, I have modified the test. Thanks for the feedback